tools/storage_mon.c

```diff
@@ -441,15 +448,7 @@ static int test_device_main(gpointer data)
 		if (is_child_runnning()) {
 			device_check = FALSE;
 		}
-		
-		if (device_count == finished_count && device_check) { 
-			/* Update the result value for the client response once all checks have completed. */
-			response_final_score = final_score;
 
-			if (!daemon_check_first_all_devices) {
```

Comment:
@oalbrigt 

Thanks for your comment.

Actually, there is no problem in leaving this if block.

However, if all check processes finish normally before check_interval, daemon_check_first_all_devices needs to be cleared immediately, so the decision needs to be made in sigchld_handler().

Otherwise, returning a correct response to the storage-mon RA monitor will be delayed until the next check_interval.

I thought that leaving this if block would cause confusion for future maintainers, so I decided not to leave it in to make it simpler.

Reply:
Makese sense.